### PR TITLE
final fixes 72

### PR DESCRIPTION
**Pull request by madhavmansuriya40**

### CODE REVIEW

File: src/app/shared/components/fy-apporver/approver-dialog/approver-dialog.component.ts [hunk 1]
```diff
@@ -102,17 +102,10 @@ export class ApproverDialogComponent implements OnInit, AfterViewInit {
...
-            return this.approverList.indexOf(approver.us.email) < 0;
-          });
-        }
-
-        if (this.from === 'ADVANCE_REQUEST') {
-          return eouc.filter(approver => {
-            return this.approverList.indexOf(approver.us.email) < 0;
-          });
-        }
+        eouc.sort((a, b) => a.us.email < b.us.email ? -1 : 1);
```
Comment (Julias0) [inline src/app/shared/components/fy-apporver/approver-dialog/approver-dialog.component.ts:105]: I dont think this sort will have any effect - is this tested?

File: src/app/shared/components/fy-apporver/approver-dialog/approver-dialog.component.ts [hunk 2]
```diff
@@ -102,17 +102,10 @@ export class ApproverDialogComponent implements OnInit, AfterViewInit {
...
+        return eouc.filter(approver => {
+          return this.approverList.indexOf(approver.us.email) < 0;
+        }).sort();
```
Comment (Julias0) [inline src/app/shared/components/fy-apporver/approver-dialog/approver-dialog.component.ts:108]: why have sort twice? Please check PRs once raising them please

File: src/app/shared/components/fy-modify-approver/modify-approver-dialog/modify-approver-dialog.component.ts [hunk 3]
```diff
@@ -123,10 +123,13 @@ export class ModifyApproverDialogComponent implements OnInit, AfterViewInit {
         return this.orgUserService.excludeByStatus(eouc, 'DISABLED');
       }),
       map(eouc => {
+        // sorting approver list
+        eouc.sort((a, b) => a.us.email < b.us.email ? -1 : 1);
```
Comment (Julias0) [inline src/app/shared/components/fy-modify-approver/modify-approver-dialog/modify-approver-dialog.component.ts:127]: why?

File: src/app/shared/components/fy-apporver/approver-dialog/approver-dialog.component.ts (hunk 1 above)
Comment (madhavmansuriya40) [reply src/app/shared/components/fy-apporver/approver-dialog/approver-dialog.component.ts:105]: yes sir, working fine

File: src/app/shared/components/fy-apporver/approver-dialog/approver-dialog.component.ts (hunk 2 above)
Comment (madhavmansuriya40) [reply src/app/shared/components/fy-apporver/approver-dialog/approver-dialog.component.ts:108]: opps, this is done by mistake

File: src/app/shared/components/fy-modify-approver/modify-approver-dialog/modify-approver-dialog.component.ts (hunk 3 above)
Comment (madhavmansuriya40) [reply src/app/shared/components/fy-modify-approver/modify-approver-dialog/modify-approver-dialog.component.ts:127]: sir need some context for `why ?`

File: src/app/shared/components/fy-apporver/approver-dialog/approver-dialog.component.ts (hunk 1 above)
Comment (Julias0) [reply src/app/shared/components/fy-apporver/approver-dialog/approver-dialog.component.ts:105]: no I meant this is wrong - sort is immutable. This line is not doing anything

File: src/app/shared/components/fy-modify-approver/modify-approver-dialog/modify-approver-dialog.component.ts (hunk 3 above)
Comment (Julias0) [reply src/app/shared/components/fy-modify-approver/modify-approver-dialog/modify-approver-dialog.component.ts:127]: dei same as above this line is not making any effect

File: src/app/shared/components/fy-apporver/approver-dialog/approver-dialog.component.ts (hunk 1 above)
Comment (madhavmansuriya40) [reply src/app/shared/components/fy-apporver/approver-dialog/approver-dialog.component.ts:105]: you mean is should be eouc = eouc.sort ?

**Review by Julias0 — APPROVED** [review]

**Pull request closed without merging**